lib/Dialect/TritonGPU/Transforms/Pipeliner/SoftwarePipeliner.cpp

```diff
@@ -123,6 +123,9 @@ struct PipelinePass : public impl::TritonGPUPipelineBase<PipelinePass> {
       auto outerLoop = dyn_cast<scf::ForOp>(forOp->getParentOp());
       int loopNumStages = getNumStagesOrDefault(forOp);
       bool pipelined = pipelineLoop(forOp, loopNumStages);
+      if (!pipelined) {
+        forOp->emitRemark("Warning: loop is not pipelined");
```

Comment:
I wonder how hard it is to provide a reason for this. Should we check to see if the loop has num_stages > 1? I am not sure if this gets executed when num_stages is 1 and if we will emit the warning always when num_stages is 1.